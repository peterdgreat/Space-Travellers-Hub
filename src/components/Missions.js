import { React } from 'react';
import { useSelector } from 'react-redux';
import MissionsList from './missionList';

export default function Missions() {
  const item = [];
  const list = useSelector((data) => data.reduceMissions.missions);
  Object.entries(list).forEach((book) => {
    const [key, value] = book;
    item.push(
      <MissionsList
        key={key}
        mission={value.mission_name}
        description={value.description}
        status={value.status}
      />,
    );
  });
  return (
    <div>
      <table className="table table-striped">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th aria-label="empty space" />
          </tr>
        </thead>
        <tbody>
          {item}
        </tbody>
      </table>
    </div>
  );
}
