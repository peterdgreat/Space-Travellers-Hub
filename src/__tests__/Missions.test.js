import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import store from '../redux/configurestore';

describe('Rockets', () => {
  it('renders without crashing', () => {
    const { getByRole } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(getByRole('table')).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
