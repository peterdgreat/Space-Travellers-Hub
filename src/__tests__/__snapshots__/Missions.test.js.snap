// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders correctly 1`] = `
<div
  className="container"
>
  <table
    className="table border table-striped w-100 mx-auto"
  >
    <thead>
      <tr>
        <th
          className="borderTable"
        >
          Mission
        </th>
        <th
          className="borderTable"
        >
          Description
        </th>
        <th
          className="borderTable"
        >
          Status
        </th>
        <th
          aria-label="empty space"
          className="borderTable"
        />
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
